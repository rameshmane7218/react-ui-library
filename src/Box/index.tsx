export { Box, BoxProps } from './Box';
