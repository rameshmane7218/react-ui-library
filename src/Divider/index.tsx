export { Divider, HrProps } from './Divider';
