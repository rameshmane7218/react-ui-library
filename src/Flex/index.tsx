export { Flex, FlexProps } from './Flex';
