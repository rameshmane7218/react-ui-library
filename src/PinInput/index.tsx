export { InputField, InputFieldProps } from './InputField';
export { PinInput, PinInputProps } from './PinInput';
export { PinInputExample } from './PinInputExample';
