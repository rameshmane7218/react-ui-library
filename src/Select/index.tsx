export { Select, SelectProps } from './Select';
export { Option, OptionProps } from './Option';
