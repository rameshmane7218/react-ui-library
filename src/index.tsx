export * from './Button';
export * from './Thing';
