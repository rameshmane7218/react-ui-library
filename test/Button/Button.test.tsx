import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Button } from '../../src';

describe('Button', () => {
  it('render without crashing', () => {
    const { getByText } = render(<Button>button</Button>);
    expect(getByText('button')).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(<Button variant="primary">button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should disabled button', () => {
    const { getByTestId } = render(
      <Button data-testid="disabled" disabled={true}>
        Button
      </Button>
    );
    expect(getByTestId('disabled')).toBeDisabled();
  });
  it('should render all variant', () => {
    const { asFragment } = render(
      <>
        <Button variant="primary">primary</Button>
        <Button variant="secondary">secondary</Button>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render all sizes', () => {
    const { asFragment } = render(
      <>
        <Button size="xs">xs</Button>
        <Button size="sm">sm</Button>
        <Button size="md">md</Button>
        <Button size="lg">lg</Button>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('onClick event fire', () => {
    let count = 0;

    const { getByTestId } = render(
      <Button data-testid="click" onClick={() => (count = 1)}>
        increment
      </Button>
    );
    fireEvent.click(getByTestId('click'));
    expect(count).toBe(1);
  });
  it('onClick event fire on disabled button', () => {
    let count = 0;

    const { getByTestId } = render(
      <Button
        data-testid="click-disabled"
        disabled={true}
        onClick={() => (count = 1)}
      >
        increment
      </Button>
    );
    fireEvent.click(getByTestId('click-disabled'));
    expect(count).toBe(0);
  });
});
