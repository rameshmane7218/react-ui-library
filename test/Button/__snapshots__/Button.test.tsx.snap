// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bcXHqe czdsTe"
  >
    button
  </button>
</DocumentFragment>
`;

exports[`Button should render all sizes 1`] = `
<DocumentFragment>
  <button
    class="sc-bcXHqe jnPCbJ"
  >
    xs
  </button>
  <button
    class="sc-bcXHqe kHDZeg"
  >
    sm
  </button>
  <button
    class="sc-bcXHqe czdsTe"
  >
    md
  </button>
  <button
    class="sc-bcXHqe cBNtft"
  >
    lg
  </button>
</DocumentFragment>
`;

exports[`Button should render all variant 1`] = `
<DocumentFragment>
  <button
    class="sc-bcXHqe czdsTe"
  >
    primary
  </button>
  <button
    class="sc-bcXHqe cbjsJS"
  >
    secondary
  </button>
</DocumentFragment>
`;
